src/black/parsing.py

```diff
@@ -208,15 +208,18 @@ def stringify_ast(node: ast.AST, depth: int = 0) -> Iterator[str]:
 
         else:
             normalized: object
-            # Constant strings may be indented across newlines, if they are
-            # docstrings; fold spaces after newlines when comparing. Similarly,
-            # trailing and leading space may be removed.
             if (
                 isinstance(node, ast.Constant)
                 and field == "value"
                 and isinstance(value, str)
             ):
+                # Constant strings may be indented across newlines, if they are
+                # docstrings; fold spaces after newlines when comparing. Similarly,
+                # trailing and leading space may be removed.
                 normalized = _normalize("\n", value)
+            elif field == "type_comment" and isinstance(value, str):
+                # Trailing spaces in type comments are removed.
```

Comment:
Interesting that we apparently do not remove trailing tabs, only spaces. That might be a bug?

In any case, this change looks good, thanks!

Reply:
Oh, we do remove tabs. Sorry, I forgot those exist.

Reply:
To be clear we actually *don't* strip trailing tabs:

```
(py311) jelle@m2mb-jelle black % black -c "x = int # type: int\t"     
x = int  # type: int\t
(py311) jelle@m2mb-jelle black % black -c "x = int # type: int   "
x = int # type: int   
error: cannot format <string>: INTERNAL ERROR: Black produced code that is not equivalent to the source.  Please report a bug on https://github.com/psf/black/issues.  This diff might be helpful: /var/folders/hh/hdb5qd2x3v51v6l0gknpfjfr0000gp/T/blk_7xu6fnsp.log
```

Reply:
Oh wait, that's not actually a tab, is it? Just a \t in the comment.

Reply:
Yeah, that's not a tab. `black -c $'x = int # type: int\t'`